src/terminal/Screen.cpp

```diff
@@ -1643,7 +1643,7 @@ void Screen<Cell>::renderImage(shared_ptr<Image const> _image,
             cell.setImageFragment(rasterizedImage, CellLocation { offset.line, offset.column });
             cell.setHyperlink(_state.cursor.hyperlink);
         };
-        moveCursorTo(_topLeft.line + unbox<int>(linesToBeRendered) - 1, _topLeft.column);
+        moveCursorTo(_topLeft.line + unbox<int>(linesToBeRendered), _topLeft.column);
```

Comment:
from [this test](https://github.com/hackerb9/vt340test/blob/main/j4james/cursor_position.sh#L3-L8) 
> Test of the text cursor position after a Sixel image is output.
 On a genuine VT340, after an image is output, the text cursor position is
 set to the same row as the top of the final sixel line (regardless of
 whether anything is output on that line). The column position remains
 unchanged, i.e. it should align with the left of the image.

@christianparpart  this change should place the ansi cursor to last sixel row, right?

Reply:
that is how i read it. any objections, @jerch?

Reply:
If that's the case I think this is ready to merge.

Reply:
I think "top of final sixel line" here means to clip the text cursor into the text row, wherever the first of the 6 pixel line of the last sixel band ended up above. So its not the last line (sixth pixel line), but the first one determining it. Note that this has a weird consequence, if the last sixel band is not perfectly mapped on text line borders, possibly cutting off lower pixel rows, or not trigger scrolling at the end of the screen.
My impl still usees the last pixel line for this, but I prolly will change that.

Reply:
So that means you may overwrite pixels in the last sixel line, if you print text just after drawing a sixel image.

Reply:
Yes thats how I understand it so far. But lets see if it gets confirmed...

Reply:
The VT340 definitely does overwrite pixels in the last sixel line, but the cases which trigger it happen very rarely as long as a text newline is sent after the sixel image. Here is an image from some of the [testing](https://github.com/hackerb9/vt340test/blob/main/sixeltests/textcursor.sh) I had done to figure out where the text cursor would be positioned based on line endings (only NL, only GNL, or neither). 
![Test of Graphics NL and Text NL after sixel images of different height](https://raw.githubusercontent.com/hackerb9/vt340test/main/sixeltests/textcursor.png)

First, I should state that it seems like the intention was for Sixel images to _not_ include a GLF at the end, but for there to be a text LF after the image if text or another image is going to be written. As @j4james pointed out to me, this allows one to show a full screen sixel image without causing scrolling. 

Second, when only LF is used at the end of the sixel data, 90% of the time there will be no overwrite. In actual practice, the glitch is much rarer than 10%. That is because people usually have images heights that are divisible by ten or a power of two or the character cell height, none of which can trigger what I call the [VT340 text cursor glitch](https://github.com/hackerb9/vt340test/blob/main/glitches.md#text-cursor-is-left-one-row-too-high-for-certain-sixel-heights). 

Personally, I think it is a programming mistake in the VT340 firmware since I cannot see anything in the documentation that would tip off an application developer to expect that behaviour. And, there does not appear to be any point to it. 

I am not sure how to emulate it, but you can predict it on the VT340. For a sixel image of height $h$, let $a=(h-1)\bmod6$ and $b=(h-1)\bmod20$, then, the text will overlap the image when $a>b$.

Reply:
@hackerb9 If you want to see some serious overlap, try this:

    echo -e '\eP1q#"60;1#1!360~\e\\'